frontend/src/app.js

```diff
@@ -31,7 +33,13 @@ const searchClient = algoliasearch(
 const search = instantsearch({
   indexName: process.env.ALGOLIA_INDEX,
   searchClient,
+  searchFunction: (helper) => {
+    TempStorage.searchError = null;
+    restoreSearchWidgets();
+    helper.search();
+  },
```

Comment:
Overriding the search function does not seem justified to me. Can you substantiate it?

Reply:
i needed a place where i can restore few things. i found this place a perfect match to achieve same.

Reply:
Thinking this over, it may be a good way of resetting the error state if you are not taking path of creating a custom widget. Yet consider my suggestion about the `render` event which I gave in another thread.

But “restoring widgets” still sounds excessive… Doesn't InstantSearch automatically take care of other widgets? I'm sure it does.

Reply:
if i am thinking correct, you are talking about event listed in this page
https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#events
i have tried with this event earlier, you can see there, render event is fired after all widgets are rendered but i need to execute this code before everything not after.